posts/multiple_index_xarray_xoak.md

```diff
@@ -0,0 +1,56 @@
+---
+author: Max Grover
+date: 2021-4-23
+tags: xarray, xoak, index
+---
+
+# Dealing with Multi-Indexing Coordinates and the Power of Xoak
+
+This week, there a post within the Zulip regarding how to deal with indexing [CAM-SE](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2017MS001257) data. The tricky part here is that is an unstructured grid, where there is only one column to the data, `ncol`.
+
+Using the [Multiple-level indexing documentation](http://xarray.pydata.org/en/stable/indexing.html#multi-level-indexing) within Xarray didn't seem to help... so what other options are there?
+
+This is a situation where using [Xoak](https://xoak.readthedocs.io/en/latest/) comes in handy. Xoak is an xarray extension, or add-on, which makes dealing with these irregular dimensions easier. It utilizes [SciPy's cKDTree adaptor](https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.cKDTree.html).
```

Comment:
"Xoak provides data structures, "indexes", that enable fast nearest-neighbour lookups on a variety of grids. It can use a variety of tree structures..."